pkg/server/server.go

```diff
@@ -274,6 +328,48 @@ func (h *handler) authenticateEndpoint(w http.ResponseWriter, req *http.Request)
 	})
 }
 
+func (h *handler) getPrivateDNSName(id string) (string, error) {
+	h.lock.Lock()
+	defer h.lock.Unlock()
+	name, ok := h.nodeNameCache[id]
+	if ok {
+		return name, nil
+	}
+	return "", errors.New("instance id not found")
+}
+
+func (h *handler) setPrivateDNSName(id string, privateDNSName string) {
+	h.lock.Lock()
+	defer h.lock.Unlock()
+	h.nodeNameCache[id] = privateDNSName
+}
+
+func (h *handler) getPrivateDNSFromEC2(id string) (string, error) {
+	privateDNSName, err := h.getPrivateDNSName(id)
+	if err == nil {
+		return fmt.Sprintf("%s%s", config.NodeNamePrefix, privateDNSName), nil
```

Comment:
For this string concatenation will probably work best and is easy to ready `config.NodeNamePrefix + privateDNSName`.